build.gradle

```
@@ -0,0 +1,24 @@
+plugins {
+    id 'java'
+}
+
+group 'Sonshal'
+version '1.0-SNAPSHOT'
+
+sourceCompatibility = 1.8
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.3.1'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.3.1'
+}
+
+sourceSets {
+    //useJUnitPlatform()
```

To enable `junit-5.x` in the project you need to specify that your project uses Junit Platform to run tests.
```suggestion
test {
    useJUnitPlatform()
}
```
And you don't need a `sourceSets` section. To enable `gradle` to see the tests you can either go for a `groovy`-path or for a `java`-path. At the moment it is something in the middle.

**Groovy Path**
* add `groovy` plugin in the `build.gradle`
* add `org.codehaus.groovy:groovy-all:2.4.15` to the list of dependencies of `test` compile scope
* rename `src/test/java` to `src/test/groovy`
* remove  `extends groovy.util.GroovyTestCase` in the class definition, since `GroovyTestCase` class is depricated and was used only in `junit-3.x`
* fix tests, because they uses assertion methods from the `GroovyTestCase` class

**Java Path**
* rename `FunctionalTest.groovy` to `FunctionalTest.java`
* remove  `extends groovy.util.GroovyTestCase` in the class definition
* fix synthetic errors

If you do everything correctly the following command will successfully build the project:
```bash
gradle clean test
```